### ISSUE TITLE
[iOS] [0.68.0-rc.4] 'react/config/ReactNativeConfig.h' file not found

### ISSUE THREAD

**Issue by AlexanderEggers** [facebook/react-native#33522]
### Description

When running the latest RC via iOS using the new architecture, I am running into the following error:

> /Users/alex/Documents/Projects/DemoFabricProject/node_modules/react-native/ReactCommon/react/renderer/scheduler/Scheduler.h:14:10: 'react/config/ReactNativeConfig.h' file not found
> 
> #include <react/config/ReactNativeConfig.h>

### Version

0.68.0-rc.4

### Output of `npx react-native info`

System:
    OS: macOS 12.3
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 1.45 GB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.14.2 - /usr/local/bin/node
    Yarn: 3.2.0 - ~/.yarn/bin/yarn
    npm: 8.5.0 - /usr/local/bin/npm
    Watchman: 2022.03.21.00 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.11.3 - /Users/alex/.rbenv/shims/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.4, iOS 15.4, macOS 12.3, tvOS 15.4, watchOS 8.5
    Android SDK:
      API Levels: 28, 30, 31, 32
      Build Tools: 29.0.2, 30.0.2, 30.0.3, 32.0.0
      System Images: android-31 | Google APIs Intel x86 Atom_64, android-31 | Google Play Intel x86 Atom_64, android-32 | Google APIs Intel x86 Atom_64, android-32 | Google Play Intel x86 Atom_64
      Android NDK: Not Found
  IDEs:
    Android Studio: 2021.1 AI-211.7628.21.2111.8193401
    Xcode: 13.3/13E113 - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.14.1 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: Not Found
    react-native: Not Found
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

### Steps to reproduce

Install project using 0.68.0-rc.4 and follow the steps provided by the upgrade helper.

**Comment by renchap** [comment]
I am seeing a similar issue:

```
❌  /Users/renchap/dev/project/packages/mobile/ios/Pods/Headers/Private/React-Codegen/react/renderer/components/rncore/EventEmitters.h:12:10: 'react/renderer/components/view/ViewEventEmitter.h' file not found

#include <react/renderer/components/view/ViewEventEmitter.h>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by renchap** [comment]
I removed my `ios/build/` and `ios/Pods/`, rebuilt everything, and it now compiles with RN 0.68.0. @AlexanderEggers have you tried this?

**Comment by baveku** [comment]
same issue, I can't build my project when upgrade to 0.68
```
 fatal error: 'react/config/ReactNativeConfig.h' file not found
```
I clean all cache, build, node_modules, Pods, and reinstall it.
But still issue, hope we can fix it soon.

**Comment by baveku** [comment]
@AlexanderEggers I found something in Reanimated, I can't build if we have Reanimated. Do you use Reanimated in your app?

**Comment by AlexanderEggers** [comment]
@baveku Yep I am using that - might indeed be related to that. Regarding reanimated generally, the PR https://github.com/facebook/react-native/commit/58a2eb7f37c2dc27ad3575618778ad5b23599b27 would unblock the new architecture for reanimated which is planned for 0.68.1. But I think reanimated generally needs to add support to the new architecture first anyway. It is strange that `fatal error: 'react/config/ReactNativeConfig.h' file not found` would be thrown in this case - seems strange to me.

**Comment by myckhel** [comment]
### Cannot build ios app with v0.68 on with or without New Architecture Enabled.
I have had patience for many days trying diff methods but still was not able to fix any and had to come up here and report.

<img width="450" alt="Screenshot 2022-04-06 at 13 37 51" src="https://user-images.githubusercontent.com/34090541/162085533-57db6ce7-5832-491b-a2ef-d7052adb0ca1.png">

**Comment by baveku** [comment]
> ### Cannot build ios app with v0.68 on with or without New Architecture Enabled.
> I have had patience for many days trying diff methods but still was not able to fix any and had to come up here and report.
> 
> <img alt="Screenshot 2022-04-06 at 13 37 51" width="450" src="https://user-images.githubusercontent.com/34090541/162085533-57db6ce7-5832-491b-a2ef-d7052adb0ca1.png">
Yes i have the same problem, so I make new project on New Template without follow migrate doc in reactnative.dev and replace file in AppDelegate.mm and reinstall all package, It's work, you can replace new Appdelegate.mm to fix it ;)

**Comment by myckhel** [comment]
> > ### Cannot build ios app with v0.68 on with or without New Architecture Enabled.
> > I have had patience for many days trying diff methods but still was not able to fix any and had to come up here and report.
> > <img alt="Screenshot 2022-04-06 at 13 37 51" width="450" src="https://user-images.githubusercontent.com/34090541/162085533-57db6ce7-5832-491b-a2ef-d7052adb0ca1.png">
> 
> Yes i have the same problem, so I make new project on New Template without follow migrate doc in reactnative.dev and replace file in AppDelegate.mm and reinstall all package, It's work, you can replace new Appdelegate.mm to fix it ;)

Thanks Your. suggestion works for me.

The content of `project.pbxproj` was the culprit, i just initialise a new react-native project and copied `project.pbxproj` into the existing project and the build works.

**Comment by StephenTANM** [comment]
> I removed my `ios/build/` and `ios/Pods/`, rebuilt everything, and it now compiles with RN 0.68.0. @AlexanderEggers have you tried this?

I was having a similar issue after enabling and disabling :fabric_enabled but this fixed it for me

**Comment by cortinico** [comment]
Closing as this has been reported as fixed + this issue was related to a RC version of RN. We now released 0.68.2 stable.

**Comment by kesha-antonov** [comment]
> Closing as this has been reported as fixed + this issue was related to a RC version of RN. We now released 0.68.2 stable.

I have this issue in 0.68.2

**Comment by kesha-antonov** [comment]
> > > ### Cannot build ios app with v0.68 on with or without New Architecture Enabled.
> > > I have had patience for many days trying diff methods but still was not able to fix any and had to come up here and report.
> > > <img alt="Screenshot 2022-04-06 at 13 37 51" width="450" src="https://user-images.githubusercontent.com/34090541/162085533-57db6ce7-5832-491b-a2ef-d7052adb0ca1.png">
> > 
> > 
> > Yes i have the same problem, so I make new project on New Template without follow migrate doc in reactnative.dev and replace file in AppDelegate.mm and reinstall all package, It's work, you can replace new Appdelegate.mm to fix it ;)
> 
> Thanks Your. suggestion works for me.
> 
> The content of `project.pbxproj` was the culprit, i just initialise a new react-native project and copied `project.pbxproj` into the existing project and the build works.

I was able to build the same way. Created new project. Opened in xcode beside my project and compared settings.
Now it builds.

**Comment by ex3ndr** [comment]
Can anyone post diffs of their project files?

**Comment by Abhishek2250** [comment]
I'm also getting the same issue:
<img width="631" alt="Screenshot 2022-05-23 at 12 11 09 PM" src="https://user-images.githubusercontent.com/97332137/169758923-cfa5144d-e50f-4b5a-8004-e4ed35b7b74f.png">

seems like it's related to react-native-reanimated
"react-native-reanimated": "^2.8.0",
"react-native": "^0.68.1",

**Comment by baveku** [comment]
> I'm also getting the same issue: <img alt="Screenshot 2022-05-23 at 12 11 09 PM" width="631" src="https://user-images.githubusercontent.com/97332137/169758923-cfa5144d-e50f-4b5a-8004-e4ed35b7b74f.png">
> 
> seems like it's related to react-native-reanimated "react-native-reanimated": "^2.8.0", "react-native": "^0.68.1",

yes, Reanimated v2 not support new Architecture,
We need to wait for Reanimated release v3,
Maybe next month, we can use it.

**Comment by sgup** [comment]
> > > > ### Cannot build ios app with v0.68 on with or without New Architecture Enabled.
> > > > I have had patience for many days trying diff methods but still was not able to fix any and had to come up here and report.
> > > > <img alt="Screenshot 2022-04-06 at 13 37 51" width="450" src="https://user-images.githubusercontent.com/34090541/162085533-57db6ce7-5832-491b-a2ef-d7052adb0ca1.png">
> > > 
> > > 
> > > Yes i have the same problem, so I make new project on New Template without follow migrate doc in reactnative.dev and replace file in AppDelegate.mm and reinstall all package, It's work, you can replace new Appdelegate.mm to fix it ;)
> > 
> > 
> > Thanks Your. suggestion works for me.
> > The content of `project.pbxproj` was the culprit, i just initialise a new react-native project and copied `project.pbxproj` into the existing project and the build works.
> 
> I was able to build the same way. Created new project. Opened in xcode beside my project and compared settings. Now it builds.

This solved it for me, I was missing the `[CP] Embed Pods Frameworks` step after manually doing the diff.

I used this build to compare without having to build a new project: https://github.com/react-native-community/rn-diff-purge/releases/tag/version%2F0.68.2

**Comment by hamzaxeros** [comment]
Same error with me too: I tried other solutions too, but there is not third-party folder exist. here is the error I guess:

```bash
ios/Pods/glog/src/base/mutex.h:105:10: fatal error: 'config.h' file not found
#include "config.h"           // to figure out pthreads support
         ^~~~~~~~~~
```

**Comment by Ananthakr** [comment]
`npx react-native-clean-project`
worked for me as well with react native@0.68.2 in both android and ios

**Comment by skantus** [comment]
Hi, I'm still facing that error with 0.68.x and 0.69.x, any idea?

![Screenshot 2022-08-07 at 15 23 23](https://user-images.githubusercontent.com/4986411/183292929-ea881ca5-774c-498a-9a47-2fa23f28851a.png)

My config is:

```
"react-native-reanimated": "2.9.1",
"react-native-screens": "3.15.0",
"react-native": "0.68.2",
"react-native-safe-area-context": "4.1.4",
```

**Comment by baveku** [comment]
> Hi, I'm still facing that error with 0.68.x and 0.69.x, any idea?
> 
> ![Screenshot 2022-08-07 at 15 23 23](https://user-images.githubusercontent.com/4986411/183292929-ea881ca5-774c-498a-9a47-2fa23f28851a.png)
> 
> My config is:
> 
> ```
> "react-native-reanimated": "2.9.1",
> "react-native-screens": "3.15.0",
> "react-native": "0.68.2",
> "react-native-safe-area-context": "4.1.4",
> ```
> 
> have you managed to solve it @baveku ?

you are using Fabric, right?
Reanimated 2 doesn't support Fabric, you can use Reanimated 3(Beta) to work with Fabric. I am still waiting for the Swmansion release Reanimated3,
Read this blog: [Announcing Reanimated 3](https://blog.swmansion.com/announcing-reanimated-3-16167428c5f7)

**Comment by skantus** [comment]
> you are using Fabric, right? Reanimated 2 doesn't support Fabric, you can use Reanimated 3(Beta) to work with Fabric. I am still waiting for the Swmansion release Reanimated3, Read this blog: [Announcing Reanimated 3](https://blog.swmansion.com/announcing-reanimated-3-16167428c5f7)

Yes, I've enabled Fabric, I'm trying to enable it in the new Architecture by using RN 0.68 and 0.69, but seems RNReanimated still can't support it, thanks got the early response.

**Comment by tomekzaw** [comment]
Hey all! In order to use Reanimated with Fabric enabled, you must first update to 0.69+.

Reanimated 3.0.0-rc.0 with Fabric enabled does not support React Native 0.68.x.

For more details, check out the comment on the issue in Reanimated repo:
* https://github.com/software-mansion/react-native-reanimated/issues/2991#issuecomment-1197886823

**Comment by baveku** [comment]
> Hey all! In order to use Reanimated with Fabric enabled, you must first update to 0.69+.
> 
> Reanimated 3.0.0-rc.0 with Fabric enabled does not support React Native 0.68.x.
> 
> For more details, check out the comment on the issue in Reanimated repo:
> 
> * [Not working with Fabric on iOS software-mansion/react-native-reanimated#2991 (comment)](https://github.com/software-mansion/react-native-reanimated/issues/2991#issuecomment-1197886823)

When does Reanimated 3 release? I saw Reanimated 3 rc0 a long time

**Comment by tomekzaw** [comment]
> When does Reanimated 3 release? I saw Reanimated 3 rc0 a long time

Reanimated 3.0.0-rc.0 supports all features of Reanimated 2 except for Layout Animations.

See this comment for more details:
* https://github.com/software-mansion/react-native-reanimated/issues/3439#issuecomment-1205382650

TL;DR We will release 3.0.0 (stable) once we add Layout Animations for Fabric.

**Comment by Manish2612** [comment]
> I removed my `ios/build/` and `ios/Pods/`, rebuilt everything, and it now compiles with RN 0.68.0. @AlexanderEggers have you tried this?

This worked for me as well.
Upgrade tool with mono-repo react-native app with multiple Xcode schemes didn't work correctly for me so I had to create fresh project.

**Comment by gus-t27** [comment]
> I removed my `ios/build/` and `ios/Pods/`, rebuilt everything, and it now compiles with RN 0.68.0. @AlexanderEggers have you tried this?

I was about to give up all hope, until I did this. Thanks you sir!

**Comment by danieldanielecki** [comment]
> I removed my `ios/build/` and `ios/Pods/`, rebuilt everything, and it now compiles with RN 0.68.0. @AlexanderEggers have you tried this?

didn't work for `React Native` version `0.70.5` coming in `Expo` version `47`. @tomekzaw, I can see the latest `rc` is `3.0.0-rc.10`. Might I ask what's your timelines to look into this?

**Comment by danieldanielecki** [comment]
Changing all `AppDelegate.m` into `AppDelegate.mm` inside `project.pbxproj` (a file with temptation not to look into) worked for me.

**Comment by atticusliu** [comment]
Still running into this issue, even with `React Native @ 0.71.3` and `Reanimated @ 2.8.0`. Removing `ios/build/`, `ios/Pods/` and cleaning the project didn't resolve it, and my `AppDelegate` inside `project.pbxproj` already has a `.mm` file extension. What am I missing?

**Comment by tomekzaw** [comment]
Hey @atticusliu, do you use the new architecture of React Native?

**Comment by tomekzaw** [comment]
> @tomekzaw yes I am - I'm running my `pod install` commands like `RCT_NEW_ARCH_ENABLED=1 bundle exec pod install` as per [the documentation](https://reactnative.dev/docs/the-new-architecture/use-app-template#enable-the-new-architecture)

Reanimated 2 does not support Fabric (the new architecture). You need to use Reanimated 3.0.0-rc.10 (or newer).

**Comment by atticusliu** [comment]
Boom, that resolved my issue. For those reading: to upgrade to Reanimated 3.0.0-rc.10 from Reanimated 2.x, you'll want to run `yarn upgrade react-native-reanimated@next`.

**Comment by baveku** [comment]
@tomekzaôm reanimated 3 still be in RC, When are we releasing it on Stable?

**Comment by mikiruano** [comment]
I see [3.0.2](https://www.npmjs.com/package/react-native-reanimated) version already available

**Comment by mikiruano** [comment]
I'll try to explain how I solved my issue with this error:

- 3.0.2 version in reanimated didn't help

Reading the react-native-config documentation I just realised Apple changed the way we have to import in AppDelegate.m

So what I did was:

Replace #import "ReactNativeConfig.h" for #import "RNCConfig.h"
Don't forget to update the whole file by changing ReactNativeConfig for RNCConfig


Happy coding :)

**Comment by joemar-tagpuno** [comment]
> @baveku I'll try to explain how I solved my issue with this error:
> 
> * 3.0.2 version in react-native-reanimated didn't help
> 
> Reading the react-native-config documentation I just realised Apple changed the way we have to import libraries in AppDelegate.m
> 
> So what I did was:
> 
> Replace **#import "ReactNativeConfig.h"** for **#import "RNCConfig.h"** Don't forget to update the whole file by changing ReactNativeConfig for RNCConfig
> 
> Happy coding :)

Thanks a lot @mikiruano , this one perfectly solved the issue.

**Comment by danieldanielecki** [comment]
> RNCConfig

It's also in the docs https://github.com/luggit/react-native-config :)

**Comment by brascene** [comment]
> @baveku I'll try to explain how I solved my issue with this error:
> 
> * 3.0.2 version in react-native-reanimated didn't help
> 
> Reading the react-native-config documentation I just realised Apple changed the way we have to import libraries in AppDelegate.m
> 
> So what I did was:
> 
> Replace **#import "ReactNativeConfig.h"** for **#import "RNCConfig.h"** Don't forget to update the whole file by changing ReactNativeConfig for RNCConfig
> 
> Happy coding :)

Exactly, thanks :)
